tests/unit_tests/main_test.cpp

```diff
@@ -112,7 +112,7 @@ int main(int argc, char** argv) {
                         if (dev.is_cpu())
                             continue;
 #endif
-#ifndef ENABLE_MKLGPU_BACKEND
+#if !defined(ENABLE_MKLGPU_BACKEND) && !defined(ENABLE_SYCLBLAS_BACKEND)
```

Comment:
If I understand correctly this is saying we will only test the SYCL-BLAS backend on Intel GPUs. Shouldn't we allow any SYCL device with our backend?

Reply:
* Yes (and no). Yes, because currently this only tests with Intel GPUs. No, because this particular statement is only about skipping Intel GPU tests.

This is because the SYCL-BLAS backend only supports Intel GPU in this initial integration. See the changes in: `include/oneapi/mkl/detail/backends_table.hpp`. 

I think the way in which SYCL-BLAS / OneMKL should support multiple backends needs discussion, and this PR is already big enough.